test/externalTests/README.md

```diff
@@ -0,0 +1,66 @@
+## Solidity external tests
+This directory contains scripts for compiling some of the popular open-source projects using the
+current version of the compiler and running their test suites.
+
+Since projects often don't use the latest compiler, we keep a fork of each of these projects
+at https://github.com/solidity-external-tests/. If changes are needed to make a project work with the
+latest version of the compiler, they're maintained as a branch on top if the upstream master branch.
+This is especially important for testing our `breaking` branch because we can not realistically expect
+external projects to be instantly compatible with a compiler version that has not been released yet.
+It also isolates us from upstream changes to some degree - their changes won't affect our test suite
+until we explicitly pull them.
```

Comment:
Actually, I'd say that's a drawback in this, not a reason for this :-).
Also the main reason for doing it is to make sure our breaking changes are sane, by verifying that those projects *can* be upgraded.

Reply:
Is it really a drawback? There were cases like the new `util-contracts` in Gnosis v2 or Colony on 0.7.0 where we could get away with just using upstream directly but I thought that having random breakage in CI due to being exposed to their changes live would not be good so I created a fork or kept the branch. Should I change that back?

Reply:
I didn't mean to say we should change anything about how it works here (although I'd argue that we should always use the upstream version directly, if it works, precisely to have CI break and actually notice it breaking immediately rather than us missing issues because we insulated us. But that's up for debate and don't we do actually do that where possible anyways?)

What I rather meant here that I wouldn't put forth isoluating us from upstream changes as reason for forking. Rather upstream not being up to date and us validating upgradability as sanity check for our breaking changes.

Reply:
I think it's good idea to have some isolation though. Upstream changes might not always be easy to adjust to and in the meantime our CI is left broken. Even more so if we start adding more of these projects. With, say, 100 projects there is a high chance of some project always being broken at any given time :)

I think that a good compromise might be to use the `master` branch of the fork (rather than e.g. `master_070`) whenever possible. In the workflow from the README, `master` is meant to be always kept in sync with the upstream without any special steps to protect against breakage on it, but if it breaks our tests it's easy to just move it back, fix the breakage and only then update again.